mojo/core/channel_posix.cc

```diff
@@ -509,6 +526,10 @@ class ChannelPosix : public Channel,
 
       std::vector<base::ScopedFD> incoming_fds;
       ssize_t read_result =
+#if defined(CASTANETS)
+          secure_connection_ ?
+          SecureSocketRecvmsg(ssl_.get(), buffer, buffer_capacity) :
```

Comment:
As you say, incoming_fds is not used and empty when secure_connection. I just wanted to make the code diff by defined(CASTANETS) as small and simple as possible.
Doing [for (auto& fd: incoming_fds)] don't burden at all (because it should be empty), so why don't you maintain code simplicity?

Reply:
I've modified what you mentioned.